packages/firebase_storage/lib/firebase_storage.dart

```diff
@@ -166,10 +186,18 @@ class StorageMetadata {
   final String contentType;
 }
 
+enum DataSource { file, bytes }
```

Comment:
Suggest to have two `StorageXxxUploadTask` classes instead. Common code might fit into a base `StorageUploadTask` class. Removing the `file` field constitutes a breaking change. If we end up doing that, we might want to reconsider the deprecation of `put` and just go with a clean break.